controller/lib/core/pinch_valve.cpp

```diff
@@ -0,0 +1,168 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "pinch_valve.h"
+#include "stepper.h"
+
+// These constants define various properties of the pinch
+// value and how we control it.  As the mechanical design
+// of the value continues to evolve these constants may
+// need to change.  Once things stabilize however, we should
+// be able to come up with a good set of values here that
+// won't need further modification.
+
+// Amplitude of the power level used for homing
+// 0.1 is 10% of maximum.  There's no need for high
+// power when homing, we're intentionally driving the
+// motor against a hard stop.
+constexpr float home_amp = 0.1f;
+
+// Velocity (deg/sec), acceleration (deg/sec/sec)
+// and distance (deg) to move during the home
+constexpr float home_vel = 60.0f;
+constexpr float home_accel = home_vel / 0.1f;
+constexpr float home_dist = 90.0f;
+
+// Amount we can move away from the homing end stop before
+// we start to touch the tube (deg)
+constexpr float home_offset = 20.0f;
+
+// This is the distance (deg) from the zero position until
+// the tube is completely shut.
+constexpr float max_move = 50.0f;
+
+// Amplitude of power level for normal operation.
+// Don't go crazy here, you can easily overheat the
+// motor driver chips and stepper motors.
+constexpr float move_amp = 0.2f;
+
+// Speed/accel for normal moves.  We want the motor
+// to be quick because the PID loop output will be
+// small move and ideally we want it to finish before
+// the next loop cycle
+constexpr float move_vel = 2000.0f;
+constexpr float move_acc = move_vel / 0.05f;
```

Comment:
All of these need to be `static` so the names are private to this file and we don't get ODR collisions on the names.